[fix] Fix parsing of styles

ingoncalves opened the pull request:
This PR fixes the translation of styles (bold, italic, underline) in Fountain to FDX. There are some cases with mixed styles (all styles put together in the same word) that were not being parsed to FDX properly. Besides that, Parenthetical elements with styles were not being parsed correctly as well.

Many tests that cover these features were not being executed, and after getting them back to work, some bugs popped out.
So this PR not only fixes bugs but also improves test coverage.

It's related to [card 3061](https://trello.com/c/iW7U4AL7).

ingoncalves committed:
[refactor] Remove unused code

ingoncalves committed:
[test] Adds byebug gem for debugging

ingoncalves committed:
[fix] Fix mixing of styles

ingoncalves committed:
[fix] Fix Parenthetical with styles

matheusefagundes commented on spec/spec_helper.rb:
```diff
@@ -1,3 +1,4 @@
+require 'byebug'
```
Pelo que entendi, você deixou o `byebug` configurado no ambiente de testes. Quando quisermos usar, basta inserir `byebug` em qualquer caso de testes para ter acesso ao debug. Se for isso, acha que cabe aqui um comentário explicando? Ou talvez no `README.md`. Só pra evitar que a gente remova esse `require` por achar que não está sendo usado em lugar nenhum.

WillianGoncalves replied:
@ingoncalves

WillianGoncalves replied:
@ingoncalves

ingoncalves replied:
Boa! Feito em ee66439c93bcb7e0a762d45439e492e6afa4e457

matheusefagundes reviewed (approved):
LGTM! Deixei uma dúvida/sugestão, mas poderia até ser feito em outro PR.

WillianGoncalves reviewed (approved):
Excelente PR! :clap:

ingoncalves committed:
[fix] Fix merging of mixing styles

Identical words with different styles are not being exported correctly
because text replacement via regex was affecting all occurrences of a
pattern at once, mixing up ccurrences improperly. This commit fixes this
behavior by replacing `gsub` with `sub`.

matheusefagundes reviewed (approved):
LGTM!
